### PR TITLE
Move size-limit to the root of Polaris

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHAT is this pull request doing?

This PR moves `size-limit` to the root of Polaris and simplifies the `size-limit.yml` workflow.

Co-authored-by: @BPScott

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .github/workflows/size-limit.yml [hunk 1]
```diff
@@ -17,9 +17,7 @@ jobs:
         with:
           node-version: 16
           cache: yarn
-      - run: yarn --frozen-lockfile
-      - run: yarn build
```
Comment (aaronccasanova) [inline .github/workflows/size-limit.yml:21]: This is already handled by the `size-limit-action` and in fact allows a custom `build_script` which we leverage to only build `@shopify/polaris` and its dependencies.

File: .github/workflows/size-limit.yml [hunk 2]
```diff
@@ -17,9 +17,7 @@ jobs:
         with:
           node-version: 16
           cache: yarn
-      - run: yarn --frozen-lockfile
```
Comment (aaronccasanova) [inline .github/workflows/size-limit.yml:20]: This is also already handled by the `size-limit-action`.

File: .github/workflows/size-limit.yml [hunk 3]
```diff
@@ -17,9 +17,7 @@ jobs:
...
-      - run: yarn build
-      - uses: andresz1/size-limit-action@v1.5.1
+      - uses: andresz1/size-limit-action@v1.7.0
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
-          directory: polaris-react/
```
Comment (aaronccasanova) [inline .github/workflows/size-limit.yml:25]: Removing `directory: polaris-react` should fix the `size-limit` failures in [PR #5803](https://github.com/Shopify/polaris/pull/5803) as `yarn build` was only being executed in `polaris-react` where what we actually need is to run a build for `polaris-react` and all its dependencies.

**Review by BPScott — APPROVED** [review]
Looks good to me. Size-limit is currently failing as it can't see a root level config when it checks out the main branch - which makes sense as there is no config in the root level package.json on main currently.

With this move to the main branch i wonder if it'd be interesting to introduce  sizelimit stats for polaris-tokens / polaris-icons too.

**Review by kyledurand — APPROVED** [review]

**Comment by aaronccasanova** [comment]
> introduce sizelimit stats for polaris-tokens / polaris-icons too.

Great idea! I added an issue (https://github.com/Shopify/polaris/issues/6049) to handle that in a follow up tasks 🙌

**Review by sam-b-rose — APPROVED** [review]

**Review by alex-page — APPROVED** [review]

**Pull request merged**